pkg/resource/graph/dependency_graph.go

```diff
@@ -69,6 +69,55 @@ func (dg *DependencyGraph) DependingOn(res *resource.State) []*resource.State {
 	return dependents
 }
 
+// DependenciesOf returns a ResourceSet of resources upon which the given resource depends. The resource's parent is
+// included in the returned set.
+func (dg *DependencyGraph) DependenciesOf(res *resource.State) ResourceSet {
+	set := make(ResourceSet)
+
+	dependentUrns := make(map[resource.URN]bool)
+	for _, dep := range res.Dependencies {
+		dependentUrns[dep] = true
+	}
+
+	if res.Provider != "" {
+		ref, err := providers.ParseReference(res.Provider)
+		contract.Assert(err == nil)
+		dependentUrns[ref.URN()] = true
+	}
+
+	cursorIndex, ok := dg.index[res]
+	contract.Assert(ok)
+	for i := cursorIndex - 1; i >= 0; i-- {
+		candidate := dg.resources[i]
+		if dependentUrns[candidate.URN] {
+			set[candidate] = true
+		}
+	}
+
+	set[dg.ParentOf(res)] = true
```

Comment:
Can we just look for this resource in the loop above? That would also let us eliminate the `ParentOf` function below.

Reply:
sure.